CMakeLists.txt

```diff
@@ -150,19 +157,38 @@ add_dependencies(examples examples_by_runtime_api_module modified_cuda_samples)
 
 install(
 	TARGETS cuda-api-wrappers
-	EXPORT cuda-api-wrappers
+	EXPORT cuda-api-wrappersExport
 	RUNTIME DESTINATION bin
 	ARCHIVE DESTINATION lib
 	LIBRARY DESTINATION lib
 	INCLUDES DESTINATION include
-	CONFIGURATIONS Release RelWithDebugInfo
 )
 
 install(
-#	EXPORT cuda-api-wrappers
 	DIRECTORY src/cuda
 	DESTINATION include
 	FILES_MATCHING REGEX "\\.(h|hpp|cuh)"
 )
 
-export(EXPORT cuda-api-wrappers FILE cuda-api-wrappers.cmake)
+install(
+	EXPORT ${PROJECT_NAME}Export
+	DESTINATION "share/cmake/${PROJECT_NAME}"
+	NAMESPACE "${PROJECT_NAME}::"
+	# In this CMake config file no dependencies are considered. But since we
+	# do not use any `find_package` call here this approach is sufficient.
```

Comment:
What about the CUDA dependency? Can we ignore that?

Reply:
Right now, everything works fine if you use the CUDA compiler `nvcc` from the CUDA toolkit. In future it might be more relevant to distinguish between a CUDA compiler and the CUDA libraries, like the runtime libraries or CUBLAS etc. I think it is right now not worth it to write CMake find modules by ourselves for the CUDA libraries. Moreover this problem is not introduced by my PR, but already exists in basically every project with CMake and CUDA libraries.